Makefile

```diff
@@ -278,7 +278,7 @@ docker-otelcontribcol:
 docker-telemetrygen:
 	GOOS=linux GOARCH=$(GOARCH) $(MAKE) telemetrygen
 	cp bin/telemetrygen_* cmd/telemetrygen/
-	cd cmd/telemetrygen && docker build --platform linux/$(GOARCH) -t telemetrygen:latest .
+	cd cmd/telemetrygen && docker build --platform linux/$(GOARCH) --build-arg="TARGETOS=$(GOOS)" --build-arg="TARGETARCH=$(GOARCH)" -t telemetrygen:latest .
```

Comment:
Is this change related?

Reply:
I can open this in a different PR, but I noticed it while building that `make docker-telemetrygen` is broken without it (at least for me):

```
$ make docker-telemetrygen
no Go files in //home/mikedame/go/src/github.com/open-telemetry/opentelemetry-collector-contrib
GOOS=linux GOARCH=amd64 make telemetrygen
make[1]: Entering directory '/home/mikedame/go/src/github.com/open-telemetry/opentelemetry-collector-contrib'
no Go files in /home/mikedame/go/src/github.com/open-telemetry/opentelemetry-collector-contrib
cd ./cmd/telemetrygen && GO111MODULE=on CGO_ENABLED=0 go build -trimpath -o ../../bin/telemetrygen_linux_amd64 \
	-tags "" .
make[1]: Leaving directory '/home/mikedame/go/src/github.com/open-telemetry/opentelemetry-collector-contrib'
cp bin/telemetrygen_* cmd/telemetrygen/
cd cmd/telemetrygen && docker build --platform linux/amd64 -t telemetrygen:latest .
Sending build context to Docker daemon   25.2MB
Step 1/10 : FROM alpine:latest as certs
 ---> 1d34ffeaf190
Step 2/10 : RUN apk --update add ca-certificates
 ---> Using cache
 ---> 958d3ffd0e84
Step 3/10 : FROM scratch
 ---> 
Step 4/10 : ARG USER_UID=10001
 ---> Using cache
 ---> 77b85f6e33ac
Step 5/10 : USER ${USER_UID}
 ---> Using cache
 ---> 22c6919286c2
Step 6/10 : ARG TARGETOS
 ---> Using cache
 ---> 742a0cf13877
Step 7/10 : ARG TARGETARCH
 ---> Using cache
 ---> 751f7303ea11
Step 8/10 : COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
 ---> Using cache
 ---> 3450c4e5656e
Step 9/10 : COPY ./telemetrygen_${TARGETOS}_${TARGETARCH} /telemetrygen
COPY failed: file not found in build context or excluded by .dockerignore: stat telemetrygen__: file does not exist
make: *** [Makefile:281: docker-telemetrygen] Error 1
```

Reply:
If it's not related, it should go to a separate PR.

Reply:
@atoulme sure, moved it to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33989